app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarMenuController.kt

```diff
@@ -226,10 +226,15 @@ class DefaultBrowserToolbarMenuController(
                 )
             }
             is ToolbarMenu.Item.SyncAccount -> {
+                val directions = if (item.signedIn) {
+                    BrowserFragmentDirections.actionGlobalAccountSettingsFragment()
+                } else {
+                    BrowserFragmentDirections.actionGlobalTurnOnSync()
+                }
```

Comment:
Can we use the account manager directly here?   That could safe a lot of work `context.components.backgroundServices.accountManager` and do the check `accountManager.authenticatedAccount() != null` ?

Reply:
Since the accountManager is tricky, I don't want to add it to the controller as well. The menu is already holding a reference, so we can check this value and pass it here. I don't think it adds any more complexity to pass the `Boolean` `signedIn` to the controller